### PR TITLE
Upgrade to React Navigation 5.0

### PR DESCRIPTION
**Pull request by thumn**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
Upgraded to React Navigation 5.0
- Created stack_navigators folder containing App Stack and Auth Stack
- Updated `ClerkLogin` to navigate to `App` instead of `CustomerLookup` upon successful login

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
N/A

### Online sources
[Upgrading React Navigation](https://reactnavigation.org/docs/upgrading-from-4.x)

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs
[#76 in customer repo](https://github.com/calblueprint/dccentralkitchen/pull/76)

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
Test flow of app from start to finish. Clerk login, customer lookup, checkout, and logout functionality should still be intact. Let me know if you run into any error screens!

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
Upgrade to Expo SDK 37

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
Mostly testing of functionality from login to logout on multiple clerk accounts. Transactions are successfully being logged in the dev Airtable.

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
N/A

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,8 +50,11 @@
     "@babel/core": "^7.0.0",
     "@expo/samples": "~3.0.3",
     "@expo/vector-icons": "^10.0.3",
-    "@react-native-community/masked-view": "^0.1.1",
+    "@react-native-community/masked-view": "^0.1.7",
     "@react-navigation/core": "^3.0.0-alpha.13",
+    "@react-navigation/drawer": "^5.4.1",
+    "@react-navigation/native": "^5.1.5",
+    "@react-navigation/stack": "^5.2.10",
     "@react-navigation/web": "^1.0.0-alpha.9",
```
Comment (anniero98) [inline package.json:58]: Nit: I think we should delete out the packages deprecated due to the upgrade such as `react-navigation/core`, `react-navigation/web`, etc https://reactnavigation.org/docs/upgrading-from-4.x/

File: navigation/DrawerContent.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            alignItems: 'flex-end',
+            padding: 16,
+          }}>
+          <Title style={{ color: 'white' }}>{this.state.clerkName}</Title>
+        </View>
+        <TouchableOpacity
+          style={{ padding: 16, paddingTop: 32 }}
+          onPress={() => Linking.openURL('http://tiny.cc/RegisterACustomer')}>
+          <Title>Register Customer</Title>
+        </TouchableOpacity>
```
Comment (wangannie) [inline navigation/DrawerContent.js]: Unrelated to this upgrade but we can take this out now
